File: src/main/scala/helpers/DataStaxCassandraGenerics.scala
package helpers

import com.datastax.oss.driver.api.core.CqlSession
import com.typesafe.scalalogging.Logger

import java.net.InetSocketAddress

trait DataStaxCassandraGenerics {
  private val logger = Logger(this.getClass)
  private val host = InetSocketAddress.createUnresolved("srv-data", 9042)

  def createSession(): CqlSession = {
      val session = CqlSession.builder()
        .addContactPoint(host)
        .withLocalDatacenter("datacenter1")
        .build()

      logger.info(f"Connected session: ${session.getName}")
      session
  }
}

File: src/main/java/com/augenz/automobiles/helpers/DataStaxCassandraGenerics.java
package com.augenz.automobiles.helpers;

import com.datastax.oss.driver.api.core.CqlSession;
import com.datastax.oss.driver.api.core.session.Session;

import java.net.InetSocketAddress;
import java.net.SocketAddress;
import java.util.logging.Logger;

public interface DataStaxCassandraGenerics {
     Logger logger = Logger.getLogger(DataStaxCassandraGenerics.class.getName());
     InetSocketAddress host = InetSocketAddress.createUnresolved("srv-data", 9042);

     static CqlSession createSession() {
         CqlSession session = CqlSession.builder().addContactPoint(host)
                 .withLocalDatacenter("datacenter1").build();

         logger.info("Connected session: " + session.getName());
         return session;
     }
}
